src/coreml.cs

```diff
@@ -381,5 +425,131 @@ interface MLCustomLayer {
 		[Export ("encodeToCommandBuffer:inputs:outputs:error:")]
 		bool Encode (IMTLCommandBuffer commandBuffer, IMTLTexture[] inputs, IMTLTexture[] outputs, [NullAllowed] out NSError error);
 	}
+
+	[Watch (5,0), TV (12,0), Mac (10,14, onlyOn64: true), iOS (12,0)]
+	[BaseType (typeof(NSObject))]
+	interface MLArrayBatchProvider : MLBatchProvider {
+
+		[Export ("array")]
+		IMLFeatureProvider[] Array { get; }
+
+		[Export ("initWithFeatureProviderArray:")]
+		IntPtr Constructor (IMLFeatureProvider[] array);
+
+		[Export ("initWithDictionary:error:")]
+		IntPtr Constructor (NSDictionary<NSString, NSArray> dictionary, [NullAllowed] out NSError error);
+	}
+
+	interface IMLBatchProvider {}
+
+	[Watch (5,0), TV (12,0), Mac (10,14, onlyOn64: true), iOS (12,0)]
+	[Protocol]
+	interface MLBatchProvider {
+
+		[Abstract]
+		[Export ("count")]
+		nint Count { get; }
+
+		[Abstract]
+		[Export ("featuresAtIndex:")]
+		IMLFeatureProvider GetFeatures (nint index);
+	}
+
+	[Watch (5,0), TV (12,0), Mac (10,14, onlyOn64: true), iOS (12,0)]
+	[Protocol]
+	interface MLCustomModel {
+
+		[Abstract]
+		[Export ("initWithModelDescription:parameterDictionary:error:")]
+		IntPtr Constructor (MLModelDescription modelDescription, NSDictionary<NSString, NSObject> parameters, [NullAllowed] out NSError error);
+
+		[Abstract]
+		[Export ("predictionFromFeatures:options:error:")]
+		[return: NullAllowed]
+		IMLFeatureProvider GetPrediction (IMLFeatureProvider inputFeatures, MLPredictionOptions options, [NullAllowed] out NSError error);
+
+		[Export ("predictionsFromBatch:options:error:")]
+		[return: NullAllowed]
+		IMLBatchProvider GetPredictions (IMLBatchProvider inputBatch, MLPredictionOptions options, [NullAllowed] out NSError error);
+	}
+
+	[Watch (5,0), TV (12,0), Mac (10,14, onlyOn64: true), iOS (12,0)]
+	[BaseType (typeof(NSObject))]
+	interface MLImageSize {
+
+		[Export ("pixelsWide")]
+		nint PixelsWide { get; }
+
+		[Export ("pixelsHigh")]
+		nint PixelsHigh { get; }
+	}
+
+	[Watch (5,0), TV (12,0), Mac (10,14, onlyOn64: true), iOS (12,0)]
+	[BaseType (typeof(NSObject))]
+	interface MLImageSizeConstraint {
+
+		[Export ("type")]
+		MLImageSizeConstraintType Type { get; }
+
+		[Export ("pixelsWideRange")]
+		NSRange PixelsWideRange { get; }
+
+		[Export ("pixelsHighRange")]
+		NSRange PixelsHighRange { get; }
+
+		[Export ("enumeratedImageSizes")]
+		MLImageSize[] EnumeratedImageSizes { get; }
```

Comment:
same, the type is only exposed as a read-only property
so there's no point in allowing it to be created, with `init`, with empty values